tests/runtests.pl

```diff
@@ -126,6 +126,7 @@ BEGIN
 
 my $noport="[not running]";
 
+my $NOLISTENPORT=2;      # port number we use for a local non-listening service
```

Comment:
I would suggest using port 47. According to [this](https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers), this port is officially reserved.

Reply:
Yeah, might be a better port. Of course [IANA having it marked as reserved](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.txt) is not any guarantee that nobody uses it for anything.

I'm also thinking we could have runtests probe the port without the use of curl code to verify it before use.

Reply:
Sure, the IANA recommendation is just recommendation. However 47 is reserved where 2 is not so I think it is much more likely that 47 will not be in use.

> I'm also thinking we could have runtests probe the port without the use of curl code to verify it before use.

My only concern with this is the cross-platform compatibility

Reply:
We already use port 2 successfully, where as we've never tried port 47...